src/main/java/com/squareup/javapoet/MethodSpec.java

```diff
@@ -187,10 +187,6 @@ public Builder addParameter(ParameterSpec parameterSpec) {
       return this;
     }
 
-    public Builder addParameter(Type type, String name, Modifier... modifiers) {
```

Comment:
Ah, so this is why you're duplicating the `CodeBlock.Builder` methods in `MethodSpec.Builder` ;-)

I'd indeed prefer dropping `ParameterSpec.of()` rather than this `addParameter` overload; that'd be consistent with `JavaPoet#add(String,TypeSpec)`.